From 5e2a7093ef46533d930da51cace64d0bb6242863 Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 8 Jan 2024 13:30:03 +0000
Subject: [PATCH 1/1] package/linux: Fixup metainfo to keep appstream composers
 happy

1. By default the .desktop file gets installed to sunshine.desktop, as
such provide the launchable tag for it to help `appstream-builder`
"find" the .desktop file due to the id (Sunshine.desktop) not matching.

Rename the id to just @PRODUCT_NAME@ as well since the .desktop suffix is
no longer required.

Provide URL as well.

Spec: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-launchable

Fixes appstream-builder generation.

2. Set the metadata_license to FSFAP. To be valid, it has to be one of
the licenses vetted by the spec, otherwise `appstreamcli-compose` will
error.

Spec: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-metadata_license

Fixes generation with `appstreamcli-compose` and thus distros which use
`appstream-generator`.
---
 packaging/linux/sunshine.appdata.xml | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/packaging/linux/sunshine.appdata.xml b/packaging/linux/sunshine.appdata.xml
index fdb6b02..a5b345f 100644
--- a/packaging/linux/sunshine.appdata.xml
+++ b/packaging/linux/sunshine.appdata.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-    <id>@PROJECT_NAME@.desktop</id>
-    <metadata_license>@PROJECT_LICENSE@</metadata_license>
+    <id>@PROJECT_NAME@</id>
+    <metadata_license>FSFAP</metadata_license>
     <project_license>@PROJECT_LICENSE@</project_license>
     <name>@PROJECT_NAME@</name>
     <url type="homepage">@CMAKE_PROJECT_HOMEPAGE_URL@</url>
@@ -11,6 +11,8 @@
             @PROJECT_LONG_DESCRIPTION@
         </p>
     </description>
+    <launchable type="desktop-id">sunshine.desktop</launchable>
+    <launchable type="url">https://localhost:47990</launchable>
     <screenshots>
         <screenshot type="default">
             <image>https://app.lizardbyte.dev/Sunshine/assets/images/AdobeStock_305732536_1920x1280.jpg</image>
-- 
2.49.0

